libraries/lib-project-file-io/SqliteSampleBlock.cpp

```diff
@@ -289,6 +298,31 @@ SampleBlockPtr SqliteSampleBlockFactory::DoCreateFromXML(
    return sb;
 }
 
+BlockSampleView SqliteSampleBlock::GetFloatSampleView()
+{
+   assert(mSampleCount > 0);
+
+   // First try to acquire the cache without locking. If that doesn't succeed,
+   // lock and try again. Can't remember how this pattern is called.
+   auto cache = mCache.lock();
+   if (cache)
+      return cache;
+
+   std::lock_guard<std::mutex> lock(mCacheMutex);
+   cache = mCache.lock();
```

Comment:
Please mention that `std::weak_ptr::lock()` guarantees atomicity, which is important to make this work without races.